### DISCUSSION TITLE
Let's Encrypt - https certificate made easy

### DISCUSSION THREAD

**Discussion by rejetto** [rejetto/hfs#346] [Show and tell]
after many hours of work, we got an easier way to get a certificate from [Let's Encrypt
](https://letsencrypt.org/)

available since [0.48.0](https://github.com/rejetto/hfs/releases/tag/v0.48.0-alpha4)

<img width="562" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/d40f80b2-eecc-4337-8e25-2667ca7652d7">

**Comment by SanokKule** [comment]
I have my domain locally overwritten in hosts file to 127.0.0.1, can you think of something for that case?
![image](https://github.com/rejetto/hfs/assets/24497821/37d54438-8874-4958-a340-d4aa1afb02e1)

**Comment by rejetto** [comment]
that happens because i'm asking the operating system to resolve the domain.
my solution is: i'm going to query 1.1.1.1 and 8.8.8.8, and if they don't answer i will fallback to the OS.

**Comment by PH5HP** [comment]
Hi...

Q: will the certificate automaticly be updated after three months?...

Kind regards,
Henk -ftp.ph5hp.nl

**Reply by rejetto** [reply]
not yet, but I think I will introduce such feature

**Comment by mcodr** [comment]
How to use this feature with duckdns.org?

**Comment by rejetto** [comment]
there's no difference with duckdns, you enter your domain where it says "domain for certificate"

**Comment by kran27** [comment]
would be nice if acme.cert could save as either .cer, .crt, or .pem, to be compatible with other software. this way hfs would spit out a valid cert for something like qBittorrent's WebUI, currently I have to use certbot.

**Comment by rejetto** [comment]
your suggestion is good and i will do it,
but please notice that you already have 2 possible solutions without waiting for my change.

solution A: select the file with qbittorrent. You'll have to force the selection interface, yes, but then it will work
<img width="278" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/71562cde-3e3b-4eda-9c17-4663aa4ea003">

solution B: you can decide the name in hfs. Just go in Admin-panel > options, and change the name here
<img width="149" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/448f36d1-d17a-4038-8ebb-486e4a23360a">

**Comment by elllectrum** [comment]
Hi,
I'm having trouble generating an SSL certificate. I keep getting an error that says "port 80 is not working on the specified domain," even though the port is open. I can connect to my server via the internet using port 80. Can anyone tell me how to solve this problem?

Additional information:

I have checked that the firewall is not blocking port 80.
I have tried restarting the server.
![image](https://github.com/rejetto/hfs/assets/64004342/d4cd2be8-8a93-4f38-a70a-b1f9c37b4179)

**Comment by rejetto** [comment]
that error is about HFS using an external server (currently located in France) to test yours.
What did you do to verify that your server is working over the internet in http?

If this test of mine proves to be unreliable, I can consider proceeding with the request anyway, and eventually provide the same error if the certiticate attempt fails.

**Comment by rejetto** [comment]
i've now fixed a bug that will make the same error appear even in the case the port was NOT really tested.
I can produce for you a preview version if you are willing to test it. If yes, tell me your operating system.

**Comment by elllectrum** [comment]
Thank you for the response! 
I access my server over the internet, including through port 80, so there shouldn't be any issues with server availability on that port. Yes, you can provide me with a preliminary version. Windows 10 pro  x64

**Comment by rejetto** [comment]
https://github.com/rejetto/hfs/files/14462618/hfs-windows.zip (52-beta5.11)
to easily install right-click on the check-updates button.
please let me know your feedback.

**Comment by elllectrum** [comment]
![image](https://github.com/rejetto/hfs/assets/64004342/fbe071b5-9369-4962-87bc-04df7d090793)

**Comment by rejetto** [comment]
brilliant!

**Comment by Meche697** [comment]
@rejetto , Just coming back to this piece of software. I self certified, but was wondering how to get the actual certificate using duckdns. My "normal" website is AAA.NORMAL.COM. I'm running HFS off of my computer at AAA.DUCKDNS.ORG. It will not allow me to create a "Let's Encrypt" certificate which would be wonderful. Any tips or tricks would be appreciated

**Reply by StiuNu** [reply]
@rejetto <img width="1858" height="433" alt="image" src="https://github.com/user-attachments/assets/c32b8ecb-3de4-4f60-ad9a-522b6005ab32" />
 Hi

I'm running in this problem when trying to get a let's encrypt certificate,

running on win

**Reply by rejetto** [reply]
@StiuNu  that means that letsencrypt tried to reach you, and failed.
if you can't be reached by them, they don't generate the certificate.
don't know more than that.

i can only add that i tried to access your HFS and it's reachable to me, but i'm not the one that genrates certificates 👀
<img width="458" height="257" alt="image" src="https://github.com/user-attachments/assets/390bb24e-dc7d-4299-91ae-2fa4cea18fca" />

**Comment by rejetto** [comment]
show me a screenshot of your problem

**Reply by pavelnil** [reply]
A few days ago there was a similar situation. After 20 attempts and disabling the firewall on the OS and the router, as well as restarting HFS, the certificate was successfully obtained. What the reason was is unclear. But there have been no changes on the firewalls for 2 months, so I don't think they are the problem. It looks like a widespread issue, a network failure. Or changes in HFS versions over 2 months.

**Comment by StiuNu** [comment]
Thank you for that info. I've spent like 6h trying to solve the issue ,
with a lot of changes to firewalls, os, ip's and  virtual machines.

I thought i'm getting to old for this job :))
